core/modules/replica/SqlResultSet.h

```diff
@@ -0,0 +1,122 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_REPLICA_SQLRESULTSET_H
+#define LSST_QSERV_REPLICA_SQLRESULTSET_H
+
+// System headers
+#include <list>
+#include <memory>
+#include <string>
+#include <vector>
+
+// Qserv headers
+#include "replica/Common.h"
+#include "replica/protocol.pb.h"
+
+// This header declarations
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+/**
+ * Structure SqlResultSet represents a result set received from a remote worker
+ * service upon a successful completion of a query against the worker database.
+ */
+struct SqlResultSet {
+
+    /// The request produced a result set
+    bool hasResult;
+
+    /// Structure Field stores a content captured from MYSQL_FIELD
+    struct Field {
+        std::string name;       /// The name of the column
+        std::string orgName;    /// The original name of the column
+        std::string table;      /// The name of the table
+        std::string orgTable;   /// The original name of the table
+        std::string db;         /// The name of the database (schema)
+        std::string catalog;    /// The catalog name (always 'def')
+        std::string def;        /// default value
+        uint32_t    length;     /// The length (width) of the column definition
+        uint32_t    maxLength;  /// The maximum length of the column value
+        uint32_t    flags;      /// Flags
+        uint32_t    decimals;   /// Number of decimals
+        int32_t     type;       /// Field type (see MySQL headers for enum enum_field_types)
+
+        /// The default c-tor is required at a presence of the explicit one
+        Field() = default;
+
+        /**
+         * Construct the object by carrying over the content of the input protocol
+         * message into the corresponding data members of the structure.
+         * 
+         * @param field
+         *   input message to be parsed
+         */
+        explicit Field(ProtocolResponseSqlField const& field);
+    };
+
+    /// A vector with field definitions from a result set,
+    /// where the number of objects in the array represents
+    /// the "width" of the result set
+    std::vector<Field> fields;
+
+    /// The row type of a result set. The number of elements in each row
+    /// must match the number of fields.
+    struct Row {
+
+        /// Values at the cells
+        std::vector<std::string> cells;
+
+        /// Flags indicating if the corresponding values of the cells represent SQL NULL
+        std::vector<bool> nulls;
```

Comment:
beware! `vector<bool>` is not your standard vector. Scott Meyers has a good writeup on this in one of his "Effective C++" books. You might/should consider a different container type.

Reply:
Him. That's interesting. I heard something on this subject long time ago, but I didn't pay attention to tis problem. And it looks like old issues are still relevant on a context of C++11 (or later versions) as per:
https://en.cppreference.com/w/cpp/container/vector_bool

I will redesign tis code to get rid of `bool`

Reply:
Thank you!

Reply:
I think it's one of those things where they can't fix it without possibly breaking existing code so they don't.